src/StackParameters.hs

```diff
@@ -199,6 +199,7 @@ stackOutputName = StackOutput <$>
     )
 
 getParameters :: MonadThrow m => Environments -> AWSAccountID -> m Parameters
+getParameters envs _ | HashMap.null envs = pure mempty
```

Comment:
Then you could just combine the case into the pattern guard:
```haskell
getParameters envs acc
  | HashMap.null envs = pure mempty
  | Just ps <- HashMap.lookup acc envs = pure ps
  | otherwise = throwM $ EnvironmentNotFound acc
```